### PR TITLE
feat: improved LiveQuery error logging with additional information

### LINKED ISSUE #7836: Syntax errors aren't logged correctly in LiveQuery triggers
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

When a syntax error is recorded in either a `beforeConnect`, `beforeSubscribe`, or `afterLiveQueryEvent` trigger, the error is logged without any useful infomation to help diagnose.

### Steps to reproduce

Create a LQ trigger with a JS error.

### Actual Outcome

```
Failed running beforeConnect for session undefined with:
     Error: {}
```

### Expected Outcome

```
Failed running beforeConnect for session undefined with:
     Error: foo is undefined
```

### Environment

Server
- Parse Server version: `master`
- Operating system: `macos`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `mongo`
- Database version: `5`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `JS`
- SDK version: `master`

### Logs

-

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Syntax Errors aren't logged with LQ triggers

Related issue: #7836

### Approach

Resolves error so that proper error message is logged in Parse Server

### TODOs before merging

- [x] Add tests
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Review by Moumouls — APPROVED** [review]
LGTM

File: spec/ParseHooks.spec.js [hunk 1]
```diff
@@ -464,9 +464,8 @@ describe('Hooks', () => {
           expect(err).not.toBe(undefined);
           expect(err).not.toBe(null);
           if (err) {
-            expect(err.code).toBe(Parse.Error.SCRIPT_FAILED);
-            expect(err.message.code).toEqual(1337);
```
Comment (mtrezza) [inline spec/ParseHooks.spec.js]: Looked strange before the change, but why had this to be  changed in this PR? And is that a breaking change?

**Review by mtrezza — APPROVED** [review]

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/5.2.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.2.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0](https://github.com/parse-community/parse-server/releases/tag/5.2.0)

**Pull request merged**